### PR TITLE
feat: implementation of first derivative for functions

### PR DESCRIPTION
**Pull request by Vivano**

### CODE REVIEW

File: calculus/api.ts [hunk 1]
```diff
@@ -48,4 +48,42 @@ export function evaluate_function(fx: string, variables : object) : APIResult {
...
+/**
+ * @param fx string expression of the function (example: x^2).
+ *          If null, empty or undefined, return InvalidParameterForFunction in error
+ * @param variable string corresponding to the variable according to which we want to compute the derivative
+ *          If null, empty or undefined, return InvalidVariableForFunction in error and the string expression of the
+ *          function in result
+ * @return json if there is no error, return the string corresponding to thr expression of the derivative in result
+ */
+export function first_simple_derivative(fx: string, variable: string) : object {
+    if (fx === undefined || fx == null){
```
Comment (QuentinRa) [inline calculus/api.ts:63]: fx === undefined is included in fx == null

**Review by QuentinRa — APPROVED** [review]

File: calculus/api.ts (hunk 1 above)
Comment (Vivano) [reply calculus/api.ts:63]: i'll fix it ;)

**Review by DataSaiyentist — APPROVED** [review]
I didn't test directly, but it seems to work well.
Tests were correctly implemented.
And the code `api.ts` seems to be correct (it seems there isn't any miss).

File: calculus/api.ts (hunk 1 above)
Comment (Vivano) [reply calculus/api.ts]: done ;)

**Pull request merged**